tests/Agent/IntegrationTests/UnboundedIntegrationTests/MySqlConnectorTests.cs

```diff
@@ -0,0 +1,201 @@
+            {
+                new Assertions.ExpectedSqlTrace
+                {
+                    TransactionName = transactionName,
+                    Sql = "SELECT _date FROM dates WHERE _date LIKE ? ORDER BY _date DESC LIMIT ?",
+                    DatastoreMetricName = "Datastore/statement/MySQL/dates/select",
+
+                    HasExplainPlan = true
+                }
+            };
+
+            var expectedTransactionTraceSegmentParameters = new List<Assertions.ExpectedSegmentParameter>
+            {
+                new Assertions.ExpectedSegmentParameter { segmentName = "Datastore/statement/MySQL/dates/select", parameterName = "sql", parameterValue = "SELECT _date FROM dates WHERE _date LIKE ? ORDER BY _date DESC LIMIT ?"},
+                new Assertions.ExpectedSegmentParameter { segmentName = "Datastore/statement/MySQL/dates/select", parameterName = "host", parameterValue = CommonUtils.NormalizeHostname(MySqlTestConfiguration.MySqlServer)},
+                new Assertions.ExpectedSegmentParameter { segmentName = "Datastore/statement/MySQL/dates/select", parameterName = "port_path_or_id", parameterValue = MySqlTestConfiguration.MySqlPort},
+                new Assertions.ExpectedSegmentParameter { segmentName = "Datastore/statement/MySQL/dates/select", parameterName = "database_name", parameterValue = MySqlTestConfiguration.MySqlDbName},
+                new Assertions.ExpectedSegmentParameter { segmentName = "Datastore/statement/MySQL/dates/select", parameterName = "explain_plan"}
+            };
+
+            var metrics = _fixture.AgentLog.GetMetrics().ToList();
+            var transactionSample = _fixture.AgentLog.TryGetTransactionSample(transactionName);
+            var transactionEvent = _fixture.AgentLog.TryGetTransactionEvent(transactionName);
+            var sqlTraces = _fixture.AgentLog.GetSqlTraces().ToList();
+
+            NrAssert.Multiple(
+                () => Assert.NotNull(transactionSample),
+                () => Assert.NotNull(transactionEvent)
+                );
+
+            NrAssert.Multiple
+            (
+                () => Assertions.MetricsExist(expectedMetrics, metrics),
+                () => Assertions.MetricsDoNotExist(unexpectedMetrics, metrics),
+                () => Assertions.TransactionTraceSegmentsExist(expectedTransactionTraceSegments, transactionSample),
+                () => Assertions.TransactionEventHasAttributes(expectedTransactionEventIntrinsicAttributes, TransactionEventAttributeType.Intrinsic, transactionEvent),
+                () => Assertions.SqlTraceExists(expectedSqlTraces, sqlTraces),
+                () => Assertions.TransactionTraceSegmentParametersExist(expectedTransactionTraceSegmentParameters, transactionSample)
+            );
+        }
+    }
+
+    [NetFrameworkTest]
+    [Collection("MySqlConnector")]
+    public class MySqlConnectorExecuteReaderTest : MySqlConnectorTestBase
```

Comment:
It's interesting to see just how similar all of these tests were, and this code structure makes that really clear!

Reply:
Yes; this could _probably_ even be pushed down into a base class for all MySQL ADO.NET tests (Connector/NET and MySqlConnector) although I'm wary of introducing dependencies between the two. All MySqlConnector methods should produce the same results, but that's not necessarily the case (long term) with MySql.Data.

Reply:
Yes, I think that it makes sense to keep the tests for these libraries separate.